File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        
        unordered_set<int> set;
        
        int answer = 0, left = 0;
        
        for (int r = 0; r < s.length(); r++) {
                while (left < r and set.find(s[r]) != set.end()) {
                    set.erase(s[left]);
                    left++;
                }
            answer = max(answer, r - left + 1);
            set.insert(s[r]);
        }
        
        return answer;
        
        
    }
};

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        charSet = set()
        
        left = 0
        answer = 0
        for r in range(len(s)):
            
            while s[r] in charSet:
                charSet.remove(s[left])
                left += 1
            charSet.add(s[r])
            
            answer = max(answer, r - left + 1)
            
        return answer
        